sys/mips/pmap.c

```diff
@@ -450,6 +452,35 @@ void pmap_set_modified(vm_page_t *pg) {
   pmap_modify_flags(pg, PTE_DIRTY, 0);
 }
 
+int pmap_emulate_bits(pmap_t *pmap, vaddr_t va, vm_prot_t prot) {
+  vm_page_t *pg;
+
+  WITH_MTX_LOCK (&pmap->mtx) {
+    paddr_t pa;
+
+    if (!pmap_extract_nolock(pmap, va, &pa))
+      return EFAULT;
+
+    pg = vm_page_find(pa);
+    pte_t pte = pmap_pte_read(pmap, va);
+
+    if ((prot & VM_PROT_WRITE) && (pte & PTE_SW_RO))
+      return EACCES;
+  }
+
+  WITH_MTX_LOCK (pv_list_lock) {
```

Comment:
Please keep the proper order of pmap::mtx and pv_list_lock. Preferred order is pv_list_lock and then pmap::mtx. Using only one convention will help us to avoid deadlocks.

Reply:
And where I do that?
I hold `pmap_t::mtx` from 458 to 468. And `pv_list_lock` from 471 to 475.
Please indicate the problematic place.